src/Illuminate/Support/Traits/Pipable.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Illuminate\Support\Traits;
+
+use Illuminate\Pipeline\Pipeline;
+use Illuminate\Support\Facades\App;
+use Throwable;
+
+trait Pipable
+{
+    /**
+     * @param array|string|callable $pipe
+     *
+     * @return $this
+     */
+    public function pipe($pipe)
+    {
+        return new class($this, $pipe)
+        {
+            private $pipable;
+
+            private $pipes;
+
+            public function __construct($pipable, $pipes)
+            {
+                $this->pipable = $pipable;
+                $this->pipes = $pipes;
+            }
+
+            public function __call($method, $params)
+            {
+                $pipeline = new Pipeline(App::getFacadeApplication());
```

Comment:
the pipeline component is not a dependency of support

Reply:
Yeah, is there any appropriate component for this trait?
I can not guest any.

Reply:
the best place for this is the foundation package tbh

Reply:
I just move the Pipable trait for "illuminate/pipeline" and since it depends on "illuminate/pipeline" I used the base `Facade` class to provide a default container if the user does not pass it.